### PR TITLE
Login persistence between sessions

### PR DESCRIPTION
**Pull request by lcflight**
- Login now persists when app is closed and reopened.

### CODE REVIEW

File: src/services/taskratchet/getMe.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ export type User = {
 
 export async function getMe(): Promise<User> {
   const response = await fetch1('me', true);
+  console.log('getMe() response', response);
```
Comment (narthur) [inline src/services/taskratchet/getMe.ts]: Remove debug logging?

File: src/services/taskratchet/useMe.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import {useQuery} from '@tanstack/react-query';
+
+import {getMe} from './getMe';
+
+export default function useMe() {
+  console.log('useMe running...');
```
Comment (narthur) [inline src/services/taskratchet/useMe.ts]: Remove debug log?

File: src/services/taskratchet/login.ts [hunk 3]
```diff
@@ -1,29 +1,49 @@
...
+            });
+          })
+          .catch(() => {
+            // No authState is set. No action needed.
+          });
+      }
+    });
+    _listenerSet = true;
+  }
+
```
Comment (narthur) [inline src/services/taskratchet/login.ts]: Delete lines 24-46?

File: src/screens/ProfileScreen.tsx [hunk 4]
```diff
@@ -25,12 +26,30 @@ export default function ProfileScreen({navigation}: Props) {
...
+        // queryClient.clear();
+        // console.log('refetching queries');
+        // await queryClient.refetchQueries();
+        console.log('resetting queries');
+        await queryClient.resetQueries();
+        console.log('queries reset');
+
+        // console.log('invalidating queries');
+        // await queryClient.invalidateQueries();
+        // console.log('queriy invalidation complete');
```
Comment (narthur) [inline src/screens/ProfileScreen.tsx]: Delete commented lines?

File: src/screens/ProfileScreen.tsx [hunk 5]
```diff
@@ -25,12 +26,30 @@ export default function ProfileScreen({navigation}: Props) {
...
+  async function handleLogoutPress() {
+    await logout()
+      .then(async () => {
+        // console.log('clearing queryClient');
+        // queryClient.clear();
+        // console.log('refetching queries');
+        // await queryClient.refetchQueries();
+        console.log('resetting queries');
+        await queryClient.resetQueries();
+        console.log('queries reset');
```
Comment (narthur) [inline src/screens/ProfileScreen.tsx]: Delete debug logs?

File: src/screens/Authenticated.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import LoginScreen from './LoginScreen';
+
+export default function Authenticated({children}: {children: React.ReactNode}) {
+  const {data, isError} = useMe();
+
+  console.log('me:', data);
+
+  // if (me.isPending) {
+  //   return <Text style={{color: 'black'}}>Loading...</Text>;
+  // }
```
Comment (narthur) [inline src/screens/Authenticated.tsx]: Delete commented code?

File: src/App.tsx [hunk 7]
```diff
@@ -1,55 +1,54 @@
...
+  return (
+    <WithSplashScreen isAppReady={true}>
+      <NavigationContainer>
+        <Authenticated>
+          <Stack.Navigator
+            initialRouteName={'HomeScreen'}
+            screenOptions={{
+              headerShown: false,
+            }}>
+            {/* <Stack.Screen name="LoginScreen" component={LoginScreen} /> */}
```
Comment (narthur) [inline src/App.tsx]: Delete commented code?

File: src/App.tsx [hunk 8]
```diff
@@ -1,55 +1,54 @@
...
-  currentUser: null,
-  setCurrentUser: (_value: userType | null) => {},
-} as {
-  currentUser: userType | null;
-  setCurrentUser: React.Dispatch<React.SetStateAction<userType | null>>;
-});
-
 function App(): JSX.Element {
-  const [currentUser, setCurrentUser] = useState<userType | null>(null);
+  // const {isFetching} = useMe();
```
Comment (narthur) [inline src/App.tsx]: Delete commented code?

**Review by narthur — APPROVED** [review]
Great work!

File: src/services/taskratchet/getMe.ts (hunk 1 above)
Comment (lcflight) [reply src/services/taskratchet/getMe.ts]: resolved

File: src/services/taskratchet/useMe.ts (hunk 2 above)
Comment (lcflight) [reply src/services/taskratchet/useMe.ts]: removed

File: src/screens/ProfileScreen.tsx (hunk 4 above)
Comment (lcflight) [reply src/screens/ProfileScreen.tsx]: deleted

File: src/screens/ProfileScreen.tsx (hunk 5 above)
Comment (lcflight) [reply src/screens/ProfileScreen.tsx]: deleted

File: src/screens/Authenticated.tsx (hunk 6 above)
Comment (lcflight) [reply src/screens/Authenticated.tsx]: deleted

File: src/App.tsx (hunk 8 above)
Comment (lcflight) [reply src/App.tsx]: removed

File: src/App.tsx (hunk 7 above)
Comment (lcflight) [reply src/App.tsx]: removed

File: src/services/taskratchet/login.ts (hunk 3 above)
Comment (lcflight) [reply src/services/taskratchet/login.ts]: removed

**Pull request merged**